pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/CompatibilityOpsRunner.java

```diff
@@ -52,6 +51,11 @@ private boolean runOps()
         System.out.println("Failure");
         break;
       }
+      /*
+       Brokers take time to update route table after a segment is created/uploaded/deleted, so sleep 1s between
+       two operations
+       */
+      TimeUnit.SECONDS.sleep(1);
```

Comment:
Adding a sleep here can make tests unreliable and susceptible to intermittent failures. You should add logic like the following:
`
ready = test_broker();
while (!ready) {
  sleep(200ms)
  ready = test_broker();
}
`
In test_broker(), you can do one of:
(1) make a call to the debug port of the broker to ensure that routing tables are built
(2) send a query to verify

Reply:
Both (1) and (2) need hardcode table name in code, which is not good in my opinion. Another idea is: in QueryOp.runOps() function, when test the first query, if the error message is "{errorCode: 410, message: "BrokerResourceMissingError"}", sleep and retry, until it succeeded or timeout.

Reply:
Wait. I am not sure why the sleep or any kind of wait is needed here at all. Let us discuss.